promql/parse.go

```diff
@@ -1000,6 +1042,12 @@ func (p *parser) checkType(node Node) (typ ValueType) {
 			p.errorf("unary expression only allowed on expressions of type scalar or instant vector, got %q", documentedType(t))
 		}
 
+	case *SubqueryExpr:
+		ty := p.checkType(n.Expr)
+		if ty != ValueTypeVector {
+			p.errorf("expected vector selector for subquery %q, got %s instead", n.String(), ty)
```

Comment:
It needs to be an instant vector, not only a vector selector

Reply:
Yes, updated the error message.